src/components/ConnectNode/ConnectNode.tsx

```diff
@@ -0,0 +1,128 @@
+import React, { FC, useState, memo, useCallback, ChangeEvent } from 'react';
+import { SidebarButton } from 'components/Sidebar';
+import DropDown from 'components/elements/DropDown';
+import { Pathway, PathwayNode } from 'pathways-model';
+import useStyles from './styles';
+import { addTransition } from 'utils/builder';
+import { usePathwaysContext } from 'components/PathwaysProvider';
+
+import { faLevelDownAlt } from '@fortawesome/free-solid-svg-icons';
+import { Button } from '@material-ui/core';
+import Tooltip from '@material-ui/core/Tooltip';
+
+interface ConnectNodeButtonProps {
+  pathway: Pathway;
+  rootNode: PathwayNode;
+}
+
+const getAncestorNodes = (
+  pathway: Pathway,
+  rootNodeKey: string,
+  currNodeKey: string
+): Array<PathwayNode> => {
+  let ancestors: Array<PathwayNode> = [];
+  let previousNodes: Array<PathwayNode> = [];
+  const currNode = pathway.nodes[currNodeKey];
+
+  for (const transition of currNode.transitions) {
+    if (transition.transition === rootNodeKey) ancestors.push(currNode);
+    else {
+      previousNodes = getAncestorNodes(
+        pathway,
+        rootNodeKey,
+        pathway.nodes[transition.transition].key ?? ''
+      );
+      if (previousNodes !== undefined && previousNodes.length > 0) {
+        if (!ancestors.some(node => node === currNode)) ancestors.push(currNode);
+        ancestors = ancestors.concat(previousNodes);
+      }
+    }
+  }
+
+  return ancestors;
+};
+
+const getConnectableNodes = (
+  pathway: Pathway,
+  rootNode: PathwayNode
+): Array<{ label: string; value: string }> => {
+  const connectableNodes: Array<{ label: string; value: string }> = [];
+  const previousNodes = getAncestorNodes(
+    pathway,
+    rootNode.key ?? '',
+    pathway.nodes['Start'].key ?? ''
+  );
+  previousNodes.push(rootNode);
+
+  for (const nodeKey of Object.keys(pathway.nodes)) {
+    const currNode = pathway.nodes[nodeKey];
+    if (!rootNode.transitions.some(transition => transition.transition === nodeKey)) {
+      if (!previousNodes.some(node => node.key === currNode.key))
+        connectableNodes.push({ label: currNode.label, value: currNode.key ?? '' });
+    }
+  }
```

Comment:
This can be cleaned up to be more readable:
```suggestion
  Object.keys(pathway.nodes).forEach((nodeKey) => {
    const node = pathway.nodes[nodeKey];
    const rootNodeConnectsToNode = rootNode.transitions.some(transition => transition.transition === nodeKey);
    const nodeIsAncestor = ancestorNodes.some(node => node.key === node.key)
    if (!rootNodeConnectsToNode && !nodeIsAncestor && node.key) 
        connectableNodes.push({ label: node.label, value: node.key});
  }
```